src/core/dbus-execute.c

```diff
@@ -3601,6 +3687,46 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (STR_IN_SET(name, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute = UINT64_MAX;
+                uint32_t quota_scale = UINT32_MAX;
+                const char *enforce_flag;
+                int quota_enforce;
+
+                r = sd_bus_message_read(message, "(tus)", &quota_absolute, &quota_scale, &enforce_flag);
+                if (r < 0)
+                        return r;
+
+                ExecDirectoryType dt;
+                if (streq(name, "StateDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_STATE;
+                else if (streq(name, "CacheDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_CACHE;
+                else if (streq(name, "LogsDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_LOGS;
+                else
+                        assert_not_reached();
+
+                quota_enforce = parse_boolean(enforce_flag);
+                if (quota_enforce < 0)
+                        return quota_enforce;
+
+                if (quota_enforce) {
+                        c->directories[dt].exec_quota.quota_absolute = quota_absolute;
+                        c->directories[dt].exec_quota.quota_scale = quota_scale;
+                        c->directories[dt].exec_quota.quota_enforce = quota_enforce;
```

Comment:
maybe move `c->directories[dt].exec_quota.quota_enforce = quota_enforce;` after the if/else expression, given you apply it effective like this on both branches